rustls/src/server/mod.rs

```diff
@@ -146,6 +146,311 @@ impl<'a> ClientHello<'a> {
     }
 }
 
+/// Building a ServerConfig in a linker-friendly way.
+///
+/// Linker-friendly: meaning unused ciphersuites, protocol
+/// versions, key exchange mechanisms, etc. can be discarded
+/// by the linker as they'll be unreferenced.
+///
+/// Example:
+///
+/// ```ignore
+/// ServerConfigBuilder::new()
+///     .with_default_ciphersuites()
+///     .with_default_kx_groups()
+///     .without_client_auth()
+///     .with_single_cert(certs, private_key)
+///     .build();
+/// ```
+///
+/// This may be shortened to:
+/// 
+/// ```ignore
+/// ServerConfigBuilder::defaults()
```

Comment:
What's the value of having `new()` and `defaults()` be different? I think I've builders usually set up sensible defaults "by default", and then methods are used to tweak the base configuration and configure those things where no meaningful default exists.

Reply:
It is explained in the comment. Perhaps the comment needs to be clarified. Use `new()` and avoid `defaults()` when you want link-time dead code optimization to drop unused code. Use `default()` when you don't care about code size.

Reply:
I guess I don't like the name `defaults()` because it's so close to and unobviously different from `Default::default()`. Also I think builders are often constructed through a `builder()` method on the type that they build. Maybe the way out here is to have a method on `ServerConfigBuilder` that directly returns `ServerConfigBuilderWithSuitesAndKxGroups`? Could be named `with_default_cipher_suites()` (or `with_default_cipher_suites_and_kx_groups()`, but that seems very wordy).

Reply:
I think maybe the simple improvement here is to just rename `defaults()` to `with_defaults()`.